text/0000-less-unwinding.md

```diff
@@ -0,0 +1,281 @@
+  strategy in the standard library needs to change. If the standard library were
+  a crates.io crate it would arguably support these options via Cargo features,
+  but without that option is this the best way to be implementing these switches
+  for the standard library?
+
+* Applications may silently revert to the wrong panic runtime given the
+  heuristics here. For example if an application relies on unwinding panics, if
+  a dependency is pulled in with an explicit `extern crate panic_abort`, then
+  the entire application will switch to aborting panics silently. This can be
+  corrected, however, with an explicit `extern crate panic_unwind` on behalf of
+  the application.
+
+# Alternatives
+[alternatives]: #alternatives
+
+* Currently this RFC allows mixing multiple panic runtimes in a crate graph so
+  long as the actual runtime is compiled with `-C panic=abort`. This is
+  primarily done to immediately reap benefit from `-C panic=abort` even though
+  the standard library we distribute will still have unwinding support compiled
+  in (compiled with `-C panic=unwind`). In the not-too-distant future however,
+  we will likely be poised to distribute multiple binary copies of the standard
+  library compiled with different profiles. We may be able to tighten this
+  restriction on behalf of the compiler, requiring that all crates in a DAG have
+  the same `-C panic` compilation mode, but there would unfortunately be no
+  immediate benefit to implementing the RFC from users of our precompiled
+  nightlies.
+
+  This alternative, additionally, can also be viewed as a drawback. It's unclear
+  what a future libstd distribution mechanism would look like and how this RFC
+  might interact with it. Stabilizing disabling unwinding via a compiler switch
+  or a Cargo profile option may not end up meshing well with the strategy we
+  pursue with shipping multiple standard libraries.
+
+* Instead of the panic runtime support in this RFC, we could instead just ship
+  two different copies of the standard library where one simply translates
+  panics to abort instead of unwinding. This is unfortunately very difficult
+  for Cargo or the compiler to track, however, to ensure that the codegen
+  option of how panics are translated is propagated throughout the rest of
+  the crate graph. Additionally it may be easy to mix up crates of different
+  panic strategies.
+
+# Unresolved questions
+[unresolved]: #unresolved-questions
+
+* One possible implementation of unwinding is via return-based flags. Much of
+  this RFC is designed with the intention of supporting arbitrary unwinding
+  implementations, but it's unclear whether it's too heavily biased towards
+  panic is either unwinding or aborting.
+
+* The current implementation of Cargo would mean that a native implementation of
+  the profile option would cause recompiles between `cargo build` and `cargo
+  test` for projects that specify `panic = 'unwind'`. Is this acceptable? Should
```

Comment:
Did you mean `panic = abort` here? (Also, did you mean "naive" instead of "native"?)

If not, then can you elaborate why specifying the same unwinding strategy as the default would force a recompile?